hack/lib/util/sync.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+#
+# This library holds sync utility functions.
+
+# os::sync::fork is responsible for synchronizing specified fork from UPSTREAM commits
+# in our vendor directory
+#
+# Arguments:
+#  - 1 - temporary directory where synchronization will take place, if not passed one will be created for you
+#  - 2 - the source repository from which take the upstream cherry-picks
+#  - 3 - the branch in source repository
+#  - 4 - the directory inside the source repository where changes are applied
+#  - 5 - the destination repository (fork) where changes are targeted
+#  - 6 - the destination repository branch
+# Returns:
+#  None
+function os::sync::fork() {
+    local tmpDir="$1"
+    local fromRepo="$2"
+    local fromBranch="$3"
+    local fromDir="$4"
+    local toRepo="$5"
+    local toBranch="$6"
+    local fromRepoName=$(basename ${fromRepo})
```

Comment:
You haven't fixed these -- you cannot mix scoping statements (`local`, `export`) or other declaration statements (`readonly`, etc) with any other expansion that has a return code. In this case if `$( basename "${fromRepo}" )` failed, the return  code would be masked by that from `local` succeeding and you would not know.

Reply:
Quote every single expansion unless you have a good reason not to:

```sh
local fromRepoName="$(basename "${fromRepo}")"
                   ^           ^           ^ ^
```